src/main/java/teammates/common/datatransfer/FeedbackSessionResultsBundle.java

```diff
@@ -76,645 +76,214 @@
-            if (order != 0) {
-                return order;
-            }
-            order = compareByResponseString(o1, o2);
-            if (order != 0) {
-                return order;
-            }
-
-            return o1.getId().compareTo(o2.getId());
-        }
-    };
+    private Comparator<FeedbackResponseAttributes> compareByTeamRecipientQuestionTeamGiver = compareByRecipientSection
+            .thenComparing(compareByNamesForRecipient)
+            .thenComparing(compareByRecipientInEmailNameTable)
+            .thenComparing(compareByQuestionNumber)
+            .thenComparing(compareByNamesForGiver)
+            .thenComparing(compareByGiverInEmailNameTable)
+            .thenComparing(compareByResponseString)
+            .thenComparing(compareByFeedbackResponseAttributeId);
 
     // Sorts by recipientTeam > question > recipientName > giverTeam > giverName
-    private Comparator<FeedbackResponseAttributes> compareByTeamQuestionRecipientTeamGiver =
-            new Comparator<FeedbackResponseAttributes>() {
-        @Override
-        public int compare(FeedbackResponseAttributes o1, FeedbackResponseAttributes o2) {
-            boolean isRecipientVisible1 = isRecipientVisible(o1);
-            boolean isRecipientVisible2 = isRecipientVisible(o2);
-            String recipientTeam1 = getTeamNameForEmail(o1.recipient).isEmpty() ? getNameForEmail(o1.recipient)
-                                                                                : getTeamNameForEmail(o1.recipient);
-            String recipientTeam2 = getTeamNameForEmail(o2.recipient).isEmpty() ? getNameForEmail(o2.recipient)
-                                                                                : getTeamNameForEmail(o2.recipient);
-            int order = compareByNames(recipientTeam1, recipientTeam2, isRecipientVisible1, isRecipientVisible2);
-            if (order != 0) {
-                return order;
-            }
-
-            order = compareByQuestionNumber(o1, o2);
-            if (order != 0) {
-                return order;
-            }
-
-            String recipientName1 = emailNameTable.get(o1.recipient);
-            String recipientName2 = emailNameTable.get(o2.recipient);
-            order = compareByNames(recipientName1, recipientName2, isRecipientVisible1, isRecipientVisible2);
-            if (order != 0) {
-                return order;
-            }
-
-            String giverTeam1 = getTeamNameForEmail(o1.giver).isEmpty() ? getNameForEmail(o1.giver)
-                                                                        : getTeamNameForEmail(o1.giver);
-            String giverTeam2 = getTeamNameForEmail(o2.giver).isEmpty() ? getNameForEmail(o2.giver)
-                                                                        : getTeamNameForEmail(o2.giver);
-            order = compareByNames(giverTeam1, giverTeam2, isRecipientVisible1, isRecipientVisible2);
```

Comment:
Quite confused as to why the 3rd and 4th parameters are about recipient and not giver. @wkurniawan07 can help?

Reply:
In my recent commit I used an abstracted comparator having `(... , isGiverVisible1, isGiverVisible2) and the tests seem to pass. Not quite sure what was the logic behind putting RecipientVisibility parameter in that function

Reply:
I'd like to believe this is a bug. Good catch!

Reply:
Great. Looking at the large number of times the same code was written, even I assumed it would be a bug. 

I have tried to reduce the code duplication and made it more readable as well 👍